setup.py

```diff
@@ -33,6 +33,10 @@
 
 tests_require = ['mockupdb>=1.4.0']
 
+extras_require = {
+    'encryption': ['pymongo[encryption]>=3.11,<4'],
```

Comment:
Do we need to add the `>=3.11,<4` here? I would hope that the version reqs are added transitively from pymongo in install_requires.

Reply:
@prashantmital thoughts on this?

Reply:
With a `setup.py` that looks like: 
```
setup(
    name='mypackage',
    version='0.1.0',
    packages=find_packages(),
    ext_modules=get_extension_modules(),
    install_requires=['pymongo==3.10'],
    extras_require={'foo': ['pymongo[srv]']},
)
```

installing the package has a rather strang output log:
```
➜   pip install -e '.[foo]'
Obtaining file:///Users/pmital/Developer/playgrounds/python-packaging
Collecting pymongo==3.10
  Downloading pymongo-3.10.0-cp37-cp37m-macosx_10_9_x86_64.whl (350 kB)
     |████████████████████████████████| 350 kB 1.6 MB/s 
Collecting pymongo[srv]
  Using cached pymongo-3.11.3-cp37-cp37m-macosx_10_6_intel.whl (414 kB)
  Downloading pymongo-3.11.2-cp37-cp37m-macosx_10_6_intel.whl (414 kB)
     |████████████████████████████████| 414 kB 1.7 MB/s 
  Downloading pymongo-3.11.1-cp37-cp37m-macosx_10_6_intel.whl (414 kB)
     |████████████████████████████████| 414 kB 2.6 MB/s 
  Downloading pymongo-3.11.0-cp37-cp37m-macosx_10_9_x86_64.whl (378 kB)
     |████████████████████████████████| 378 kB 3.2 MB/s 
  Downloading pymongo-3.10.1-cp37-cp37m-macosx_10_9_x86_64.whl (350 kB)
     |████████████████████████████████| 350 kB 3.8 MB/s 
Collecting dnspython<2.0.0,>=1.16.0
  Using cached dnspython-1.16.0-py2.py3-none-any.whl (188 kB)
Installing collected packages: pymongo, dnspython, mypackage
  Attempting uninstall: mypackage
    Found existing installation: mypackage 0.1.0
    Uninstalling mypackage-0.1.0:
      Successfully uninstalled mypackage-0.1.0
  Running setup.py develop for mypackage
Successfully installed dnspython-1.16.0 mypackage pymongo-3.10.0
➜   pip list 
Package    Version Location
---------- ------- ----------------------------------------------------
Cython     0.29.22
dnspython  1.16.0
mypackage  0.1.0   /Users/pmital/Developer/playgrounds/python-packaging
pip        21.0.1
pymongo    3.10.0
setuptools 47.1.0
wheel      0.36.2
```

for some reason, it downloads all the versions between 3.10 and latest but ends up only installing the one in `install_requires`. So @ShaneHarvey does seem to be correct in that `pip` honors the `install_requires` version. That being said, it might result in longer install times if there are a lot versions to download. 

Specifying the version in `extras_require` eliminates the extra downloads:
```
➜  pip --no-cache-dir install -e '.[foo]'
Obtaining file:///Users/pmital/Developer/playgrounds/python-packaging
Collecting pymongo==3.10
  Downloading pymongo-3.10.0-cp37-cp37m-macosx_10_9_x86_64.whl (350 kB)
     |████████████████████████████████| 350 kB 2.1 MB/s 
Requirement already satisfied: dnspython<2.0.0,>=1.16.0 in /Users/pmital/.pyenv/versions/3.7.10/envs/packaging/lib/python3.7/site-packages (from pymongo==3.10->mypackage==0.1.0) (1.16.0)
Installing collected packages: pymongo, mypackage
  Running setup.py develop for mypackage
Successfully installed mypackage pymongo-3.10.0
```

@ShaneHarvey how would you like to proceed?

Reply:
Oh well. Let's keep `>=3.11,<4`. We'll just need to ensure those two reqs never drift apart.

Reply:
Yes let's do your single-source option.